Configure

```diff
@@ -1737,7 +1744,7 @@ while (<IN>)
 		s/^AR=\s*ar/AR= $ar/;
 		s/^RANLIB=.*/RANLIB= $ranlib/;
 		s/^RC=.*/RC= $windres/;
-		s/^MAKEDEPPROG=.*$/MAKEDEPPROG= $cc/ if $predefined{__GNUC__} >= 3;
+		s/^MAKEDEPPROG=.*$/MAKEDEPPROG= $makedepprog/ if defined($makedepprog);
```

Comment:
When made remark about "this" and "that" ifs, I mean that you'd replace `$predefined{__GNUC__} >= 3` with `if $makedepprog` in *both* cases. This is because we collect pre-defines from $cross_compile_prefix$cc, not just $cc. Note that workaround is likely to be needed even with `__APPLE_CC__`, because that's what's used even for ios targets, and you would set cross-compile prefix...